src/cli/init.ts

```diff
@@ -74,13 +74,58 @@ export const initCommand = new Command()
     const context: object = { circuitName, circuitType };
 
     // Handle individual circuit types.
-    // Gnark.
-    if (circuitType === "gnark") {
+    if (circuitType === "circom") {
+      // Circom.
+      const templateName = await input({
+        message: "Circom Template Name:",
+        default: circuitName
+          .replace(/[^a-zA-Z0-9_]]+/g, "_")
+          .replace(/^[^a-zA-Z_]/g, "_")
+          .replace(/_+/g, "_"),
+        validate: (input): boolean | string => {
+          if (input.length === 0) {
+            return "You must specify a template name.";
+          }
+          if (!/^[a-zA-Z_][a-zA-Z0-9_]*$/.test(input)) {
+            return (
+              "Template names must begin with a letter or underscore and only be followed by " +
+              "alphanumeric characters or underscores."
+            );
+          }
+          return true;
+        },
+      });
+      const provingScheme: "groth16" = await select({
+        message: "Proving Scheme:",
+        default: "groth16",
+        choices: [{ name: "Groth16", value: "groth16" }],
+      });
+      const curveName: "bn254" = await select({
+        message: "Curve Name:",
+        default: "bn254",
+        choices: [{ name: "BN254", value: "bn254" }],
+      });
+      const witnessCompiler: "c++" | "wasm" = await select({
+        message: "Witness Compiler:",
+        default: "c++",
+        choices: [
+          { name: "C++", value: "c++" },
+          { name: "Wasm", value: "wasm" },
+        ],
+      });
+      Object.assign(context, {
+        curveName,
+        provingScheme,
+        templateName,
+        witnessCompiler,
+      });
+    } else if (circuitType === "gnark") {
+      // Gnark.
       const packageName = await input({
         message: "Go Package Name:",
         default: circuitName
           .replace(/[^a-zA-Z0-9]/g, "")
-          .replace(/^[^a-z]*/, ""),
+          .replace(/^[^a-z]*/g, ""),
```

Comment:
Also unrelated, but we need to remove all of these characters at the start of the Gnark package names when there are more than one.